src/main.js

```diff
@@ -19,8 +19,10 @@ if (process.platform === 'win32') {
   var cmd = process.argv[1];
   if (cmd === '--squirrel-uninstall') {
     var AutoLaunch = require('auto-launch');
+    //BUILD
     var appLauncher = new AutoLaunch({
-      name: 'Mattermost'
+      name: 'Mattermost',
+      isHidden: true
```

Comment:
I misunderstood the changelog. This part is called when uninstalling the application. So you should use `isHidden` in `settings.jsx` which enables auto-start. Then, possibly `main.js` should handle the `--hidden` flag.